docs/_posts/2022-05-06-writing-cli-tools-with-typescript.md

```diff
@@ -0,0 +1,187 @@
+2. **popularity** (TypeScript was ranked 7th in the [2021 Stack Overflow Developer Survey](https://insights.stackoverflow.com/survey/2021#section-most-popular-technologies-programming-scripting-and-markup-languages) with JavaScript being 1st),
+3. **tooling choice** (the Node.js ecosystem provides [quite a lot of CLI libraries](https://openbase.com/categories/js/best-javascript-cli-libraries)),
+4. **developer friendliness** (TypeScript is a scripted language but also has an advanced type system and static type checking which helps writing high-quality code),
+5. **performance** (since TypeScript code is, in the end, typically run by Node.js, it is in general sufficiently performant thanks to the [v8](https://v8.dev/) engine).
+
+Nonetheless, a typically non-trivial part of any TypeScript project is the set up. Writing a CLI with e.g. Python and the [Click](https://click.palletsprojects.com/en/8.1.x/) library requires little more than writing a file with the [main function](https://docs.python.org/3/library/__main__.html#packaging-considerations). The higher setup complexity of TypeScript is of course due to the nature of TypeScript being a superset of JavaScript. To run TypeScript code, you typically transpile it to JavaScript and then execute the JavaScript code with Node.js. (Yes, even [ts-node](https://github.com/TypeStrong/ts-node) does that.) So, before running a program written in TypeScript, we need to transpile it to JavaScript first. This requires tooling and therefore setup.
+
+Preferably, we also want to have a single, packaged binary as the result of building our project, rather than a script file, to reduce installation overhead on part of the user. 
+
+To assure high code quality we also typically want to include code formatting, linting, and tests.
+
+Preferably, we also want to have a single, packaged binary as the result of building our project rather than a script file to reduce installation overhead on part of the user. 
+
+To assure high code quality we also typically want to include code formatting, linting, and tests.
+
+## Basic development flow
+
+To hit the ground running, I created a starter project which already contains all required boilerplate and dependencies for running, formatting, linting, testing the code and packaging it to a binary. 
+
+> ℹ️ The starter project can be found on my [GitHub](https://github.com/fkurz/typescript-cliutil-starter) and via [npm](https://www.npmjs.com/package/@fkurz/typescript-cliutil-starter/). 
+
+```bash
+git clone git@github.com:fkurz/typescript-cliutil-starter.git my-cli
+cd my-cli 
+npm i
+```
+
+Our starter project uses 
+
+- the [*commander*](https://www.npmjs.com/package/commander) package for argument parsing and executing commands, 
+- code linting and formatting with [eslint](https://eslint.org/) and [prettier](https://prettier.io/), 
+- [Jest](https://jestjs.io/) as test runner and assertion library, and 
```

Comment:
Not sure why you always have an `,` before `and`. It's also common to not write `and` at all but just a list, but both feels odd to me, doesn't it?

Reply:
It's called an _Oxford comma_ (https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/). 

Writing a list this way was standard at my university. May be awkward for some, I agree. What's your suggestion?

Reply:
Well, than keep your style of writing :)

Reply:
> Well, **then** keep your style of writing :)

FTFY ;)